lib/src/components/NxScrollRender/NxScrollRender.tsx

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { ReactElement, useCallback, useEffect, useLayoutEffect, useMemo, useRef, useState } from 'react';
+import { always, clamp, dec, identity, inc, isNil } from 'ramda';
+import useResizeObserver from '@react-hook/resize-observer';
+import { useThrottleCallback } from '@react-hook/throttle';
+
+import { Props } from './types';
+
+// intended to wrap division operations that may result in NaN, returns zero instead in that case
+const divOrZero = (x: number, y: number) => x === 0 || y === 0 ? 0 : x / y;
+
+const DEFAULT_INITIAL_CHILD_COUNT = 40;
+
+export default function NxScrollRender({ children, reuseChildren, initialChildCount, spacerEl }: Props) {
+  const fullParent = children,
+
+      childArray = useMemo(() => React.Children.toArray(fullParent.props.children) as ReactElement[],
+          [fullParent.props.children]),
+
+      childCount = childArray.length,
+
+      parentRef = useRef<HTMLElement>(null),
+      leadingSpacerRef = useRef<HTMLElement | null>(null),
+      leadingSpacerRefFn = (e: HTMLElement | null) => { leadingSpacerRef.current = e; },
+      trailingSpacerRef = useRef<HTMLElement | null>(null),
+      trailingSpacerRefFn = (e: HTMLElement | null) => { trailingSpacerRef.current = e; },
+
+      [parentHeight, setParentHeight] = useState<number | null>(null),
+      [childHeight, setChildHeight] = useState<number | null>(null),
+      normalizedChildHeight = childHeight ?? 0,
+      renderedChildCount = parentHeight == null || childHeight == null ?
```

Comment:
Out of curiosity, is there a reason using `==` here instead of `===`?

Reply:
> Out of curiosity, is there a reason using == here instead of ===?

`== null` is pretty much the only good use case for `==`.  It returns true if the value is `null` or `undefined`, and `false` otherwise.

Reply:
Thanks for the explanation.